### ISSUE TITLE
Compiler is unable to find funit.mod

### ISSUE THREAD

**Issue by n-claes** [Goddard-Fortran-Ecosystem/pFUnit#270]
I'm not sure if this is an actual bug or if there were some changes in using pFUnit between the most recent version and the one from two months ago, but a compilation that worked fine with previous versions suddenly errors out.

Some background: we're using pFUnit for unit testing of our new Legolas code, and we have set this up on GitHub Actions. The runner actually caches the pFUnit installation and restores it so it doesn't have to recompile every time a commit is made. However, a few weeks have passed since the last push to our master branch, such that the cached version expired and a fresh clone was installed. When compiling our project, it suddenly errors out in the `make` stage with
```
use funit, only: pFUnit_throw => throw

Fatal Error: Cannot open module file ‘funit.mod’ for reading at (1): No such file or directory
```
See for example our test log [here](https://github.com/n-claes/legolas/runs/1777444973?check_suite_focus=true).
This error can be reproduced locally as well on both macOS and linux, and resetting the pFUnit clone to commit 2c96c37 works like a charm. 

Were there some changes in how the `funit` module should be used between that version and more recent ones?
For the moment we fixed our pFUnit clone to the aforementioned commit so this isn't urgent.

Thanks for the great effort in developing/maintaining this tool, it is exactly what we needed for our unit tests :)

**Comment by ZedThree** [comment]
@n-claes Please could you see if #268 fixes the problem for you?

**Comment by kurtsansom** [comment]
version 4.1.14 appears to have the same issue.

``` 
/heat/build/tests/unit_tests/core/utility/test_class_distribution.F90:6:6:
    use funit
       1
 Fatal Error: Can't open module file 'funit.mod' for reading at (1): No such file or directory
 compilation terminated.
```

**Comment by n-claes** [comment]
So I tested the fix proposed in #268 and that partially solves the issue. I added that code snippet to our build script and linked `PFUNIT_LIBRARIES` explicitly in the `LINK_LIBRARIES` part of `add_pfunit_ctest`, which resulted in a proper compilation of all files without the above error. Instead it now fails at the very end when make tries to link the Fortran executable:
```
ld: library not found for -lfunit
```
Not sure if this is due to the above fix or an issue with our build script, since all other libraries we use do link properly.
